src/lbaf/Execution/lbsAlgorithmBase.py

```diff
@@ -18,7 +18,12 @@ def __init__(self, work_model, parameters: dict, lgr: Logger, qoi_name: str=''):
             parameters: a dictionary of parameters
             qoi_name: optional additional QOI to track"""
 
-        # Assign logger to instance variable
+        # Assert that a logger instance was passed
+        if not isinstance(lgr, Logger):
+            lgr().error(
+                f"Incorrect provided number of algorithm iterations: {self.__n_iterations}")
+            sys.excepthook = exc_handler
+            raise SystemExit(1)
```

Comment:
I don't understand this. If a logger wasn't passed, then use the logger (that wasn't passed) to print an error about the number of iterations?

Reply:
Great catch! Wrong message indeed.  Fixed